pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/generator/PinotTaskGenerator.java

```diff
@@ -99,4 +99,10 @@ default void nonLeaderCleanUp(List<String> tableNamesWithType) {
   default String getMinionInstanceTag(TableConfig tableConfig) {
     return CommonConstants.Helix.UNTAGGED_MINION_INSTANCE;
   }
+
+  /**
+   * Performs necessary validations for the given task type.
+   */
+  default void validateTaskConfigs(TableConfig tableConfig, Map<String, String> taskConfigs) {
```

Comment:
May be add doc or rename the param to indicate that taskConfigs here is the input taskConfigs that needs to be validated.